### PR TITLE
[SYCLomatic #1311]Add behavior test for incorrect compilation database generation of intercept-build tool

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: behavior_tests/src/process_optf_option/use_optf_option_case/test1.cu [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// ====------ test2.cu---------------------------------- *- CUDA -* ----===////
```
Comment (ShengchenJ) [inline behavior_tests/src/process_optf_option/use_optf_option_case/test1.cu]:
```suggestion
// ====------ test1.cu---------------------------------- *- CUDA -* ----===////
```

File: behavior_tests/src/process_optf_option/use_optf_option_case/Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+all:test
+
+test:test1.o test2.o
+
+test1.o:test1.cu
+	nvcc -c test1.cu --options-file=./includes1.rsp -optf=./includes2.rsp --options-file ./includes3.rsp -optf=./includes4.rsp
```
Comment (ShengchenJ) [inline behavior_tests/src/process_optf_option/use_optf_option_case/Makefile]: Why add "-optf=./includes4.rsp" as the -optf=./includes2.rsp  already has covered same behavior?

**Review by ShengchenJ — CHANGES_REQUESTED** [review]

File: behavior_tests/src/process_optf_option/use_optf_option_case/Makefile (hunk 2 above)
Comment (tomflinda) [reply behavior_tests/src/process_optf_option/use_optf_option_case/Makefile]: "-optf=./includes4.rsp" should be "-optf ./includes4.rsp", will change it in the update PR.

File: behavior_tests/src/process_optf_option/use_optf_option_case/test1.cu (hunk 1 above)
Comment (tomflinda) [reply behavior_tests/src/process_optf_option/use_optf_option_case/test1.cu]: Okay

**Review by ShengchenJ — APPROVED** [review]

**Pull request merged**